File: python/leetcode/MaxIncreaseToKeepCitySkyline/MaxIncreaseKeepingSkyline.cs
using System;
using Newtonsoft.Json;

namespace ConsoleAppDemoOne
{
    public class Solution {
        public int MaxIncreaseKeepingSkyline(int[][] grid) {
            int n = grid.Length;
            int[] rowMax = new int[n];
            int[] colMax = new int[n];
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    rowMax[i] = Math.Max(rowMax[i], grid[i][j]);
                    colMax[j] = Math.Max(colMax[j], grid[i][j]);
                }
            }
            int ans = 0;
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    ans += Math.Min(rowMax[i], colMax[j]) - grid[i][j];
                }
            }
            return ans;
        }
        
        public static void Main(string[] args)
        {
            var gridInput = Console.ReadLine();
            var grid = JsonConvert.DeserializeObject<int[][]>(gridInput ?? String.Empty);
            Console.WriteLine(new Solution().MaxIncreaseKeepingSkyline(grid));
        }
        
    }
}

File: python/leetcode/MaxIncreaseToKeepCitySkyline/MaxIncreaseKeepingSkyline.cpp
#include <iostream>
#include <vector>
#include ".\lib\json-3.9.1\single_include\nlohmann\json.hpp"

using json = nlohmann::json;
using namespace std;

class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int n = grid.size();
        vector<int> rowMax(n);
        vector<int> colMax(n);
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                rowMax[i] = max(rowMax[i], grid[i][j]);
                colMax[j] = max(colMax[j], grid[i][j]);
            }
        }
        int ans = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                ans += min(rowMax[i], colMax[j]) - grid[i][j];
            }
        }
        return ans;
    }
};


int main() {
    string gridStr;
    getline(cin, gridStr);
    auto gridJson = json::parse(gridStr);
    vector<vector<int>> grid = gridJson;
    cout << Solution().maxIncreaseKeepingSkyline(grid) << endl;
}
